api/src/main/java/io/lumeer/engine/api/data/DataStorage.java

```diff
@@ -136,87 +137,62 @@ default void connect(final StorageConnection connection, final String database,
    void createOldDocument(final String collectionName, final DataDocument document, String documentId, int version) throws UnsuccessfulOperationException;
 
    /**
-    * Reads the specified document in given collection by its id.
+    * Reads the specified document in given collection by filter.
     *
     * @param collectionName
     *       the name of the collection where the document is located
-    * @param documentId
-    *       the id of the read document
+    * @param filter
+    *       the filter to identify document
     * @param attributes
     *       list of attribute names
     * @return the DataDocument object representing the read document containg only specified attributes
     */
-   DataDocument readDocumentIncludeAttrs(final String collectionName, final String documentId, final List<String> attributes);
-
-   /**
-    * Reads the specified document in given collection by its id.
-    *
-    * @param collectionName
-    *       the name of the collection where the document is located
-    * @param documentId
-    *       the id of the read document
-    * @return the DataDocument object representing the read document
-    */
-   DataDocument readDocument(final String collectionName, final String documentId);
+   DataDocument readDocumentIncludeAttrs(final String collectionName, final String filter, final List<String> attributes);
 
    /**
-    * Reads the old document in given collection by its id and version.
+    * Reads the specified document in given collection by filter.
     *
     * @param collectionName
     *       the name of the collection where the document is located
-    * @param documentId
-    *       the id of the document
-    * @param version
-    *       the version of document
+    * @param filter
+    *       the filter to identify document
     * @return the DataDocument object representing the read document
     */
-   DataDocument readOldDocument(final String collectionName, final String documentId, final int version);
+   DataDocument readDocument(final String collectionName, final String filter);
```

Comment:
I did this refactor due to lot of duplicated methods so I think that approach is better.